sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py

```diff
@@ -68,7 +71,19 @@ def _copy(src: Path, dst: Path) -> None:
         if src.is_file():
             _general_copy(src, dst)
         else:
-            shutil.copytree(src, dst, ignore=shutil.ignore_patterns("__pycache__"))
+            # use os.walk to replace shutil.copytree, which may raise FileExistsError
+            # for same folder, the expected behavior is merging
+            # ignore will be also applied during this process
+            ignore_file = InternalComponentIgnoreFile()
```

Comment:
will it be better if we make the ignore file a property of additional includes or parameter in case we need to customize it later?

Reply:
will make it a property of additional includes comes from internal component

Reply:
ignore file has some logic related to path, and for internal component, we have some logic changing code path (to temp folder), so I think maybe it's better not keep a ignore file as property, but its rules.
I create a property `_ignores` and add some logic in `InternalComponentIgnoreFile` to include extra ignores
and when need it, create an ignore file with property `_ignores`.